### PR TITLE
Week3 finished.

### PR DESCRIPTION
**Pull request by MarkNou**
# Submitting Week 3 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ X] Coding Challenges
- [ X] Interview Questions
- [ X] React Debugging

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here: N/A

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -1,31 +1,121 @@
...
+
+
+
+// // //Pseudo Code:
+// // //Create a function called fibSeq (short for Fibonacci Sequence)
+// // //Create a parameter called arrayLength that will hold the length of an array, which will be a number.
+// // //Create a variable called 'fib' that will hold the array of numbers. It will start off having [1, 1] since the array needs to be greater than two numbers. Also the fibonacci sequence always starts off with 1 and 1.
+// // //Create a for loop. Since we already have the first two indexes in our fib array, we start the loop at index 2 (the third value). We set the length of the array to our parameter arrayLength. We then use i++ to itertate over value at an increment of 1.
+// // //Using the arrayLength, we iterate through that many times. fib[i-1] + fib[i-2] will add the previous and next indexs together into an array.
+// // //Console.log the function with a parameter and return an array with the fibonacci sequence.
```
Comment (Genemichael25) [inline code-challenges.test.js]: Great job on this one with the pseudocode and the function looks great! No need for the console.log , your test is set up correctly and if it passes you know you are good to go. 👍🏽

File: code-challenges.test.js [hunk 2]
```diff
@@ -1,31 +1,121 @@
...
+
+// // // --------------------2) Create a function that takes in an object and returns an array of the values sorted from least to greatest.
+// // // Hint: Check out this resource: Object.values() https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values
+
+// // // a) Create a test with expect statements for each of the variables provided.
+
+describe("objectSorter", () => {
+  it("takes in an object and returns an array of the values sorted from least to greatest", () => {
+    expect(
+      objectSorter({
```
Comment (Genemichael25) [inline code-challenges.test.js:88]: You can use the variables here in your expect. For example, expect(objectSorter(studyMinutesWeek1)) and the rest that follows

File: code-challenges.test.js [hunk 3]
```diff
@@ -49,12 +139,54 @@ const studyMinutesWeek2 = {
...
-// b) Create the function that makes the test pass.
+// // b) Create the function that makes the test pass.
+
+
+
+// // Create a function called objectSorter that takes in an object and returns an array of the values sorted from least to greatest
+// // Create a parameter called 'data' as a placeholder for the object we will sort. I used 'data' instead of 'object' since the method we will be using already contains the word 'object'. This will prevent any confusion
+// // Create a variable called 'sorted' to hold the logic
+// // Use 'Object.value()' to extract the values from the keys in an object
+// // Chain .sort() at the end to sort the values from least to greatest
```
Comment (Genemichael25) [inline code-challenges.test.js]: I would add the way we will ensure it sorts from least to greatest. Looks great!!! 💯

File: interview-questions.md [hunk 4]
```diff
@@ -8,48 +8,65 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
+Researched answer: A React component is a reusable building block for creating UI. A component is a self-contained block of code, that can be plugged in and used with other components. Components can recieve data through props, which allow them to operate like functions with arguments. As data changes, components will also dynamically display those changes while still maintaining their structure.
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: State values are used to store information that can be updated and changed. By using useState(), developers can use a state variable and a setter to make their UI interactable. Props are a way for components to communicate to one another. Components in different files can reference props in other files in React. The biggest difference is state values are used for performing specifc logic while props are used for communicating tasks.
```
Comment (Genemichael25) [inline interview-questions.md:42]: Props are great for passing data and functions from parent to child components.

File: interview-questions.md [hunk 5]
```diff
@@ -8,48 +8,65 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 Researched answer:
 
+DOM is a programming interface for websites. DOM allows developers to interact with the structure of a website via JavaScript. It allows developers to respond to user events and create dynamic, interactive webpages. 
+
 6. STRETCH: Which is the difference between a div tag and a span tag?
 
-Your answer:
+Your answer: I have only used a div tag, I've never heard or used a span tag. A div tag is used as a container in HTML. Its often used as a block/container to hold other HTML tags. It compartmentalizes data that is relative to each other. For example, you could use a <div></div> to hold information about a product with a heading <h3></h3>, description <p></p>, and features <li></li>.
+
+Researched answer: the div and span tag are both used in HTML to organize and display content. Div is a block-level element that takes up an entire container and starts on a new line. Span is an inline element that takes up only the amount of space it needs, and does not start on another line.
```
Comment (Genemichael25) [inline interview-questions.md:58]: Exactly! Nice follow-up on the span tag.

File: react-debugging.html [hunk 6]
```diff
@@ -8,87 +8,94 @@
...
+        width: "70px",
+        height: "95px",
+        margin: "10px",
+        padding: "10px"
+      }
         // No need to change anything ABOVE this line ^
 
+        
+        //Errors fixed:
+        //First button needed to call a function for 'onClick='. Since the html text for that button is 'Click to Draw a Card', I assigned it the {drawCard} function.
```
Comment (Genemichael25) [inline react-debugging.html:65]: Great job finding those bugs! 🐛

File: interview-questions.md [hunk 7]
```diff
@@ -8,48 +8,65 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
 3. What is a React component?
 
-Your answer:
+Your answer: I view React components as templates for information. Creating containers of data can become extremley repetitive and redundant, and it also requires a lot of line of code which can make your code look messy. React components allow you to make "blank" templates and call them over and over again on your page. The data and logic can be easily interchangable, making them highly dynamic. 
 
-Researched answer:
+One example I like to think of when referring to React components is a media player on a streaming service such as Spotify. The media player has a album picture, play/pause button, and a back/next button. By creating just one component for this template, you can then pass the information of millions of songs into it.
+
+Researched answer: A React component is a reusable building block for creating UI. A component is a self-contained block of code, that can be plugged in and used with other components. Components can recieve data through props, which allow them to operate like functions with arguments. As data changes, components will also dynamically display those changes while still maintaining their structure.
```
Comment (Genemichael25) [inline interview-questions.md:38]: Love the examples in this answer and definitely can be thought of as a building block. 💯

File: code-challenges.test.js [hunk 8]
```diff
@@ -65,3 +197,23 @@ const accountTransactions3 = []
 // Expected output: []
 
 // b) Create the function that makes the test pass.
+
```
Comment (Genemichael25) [inline code-challenges.test.js:200]: Missing pseudocode.

**Review by Genemichael25 — CHANGES_REQUESTED** [review]
Awesome work on this assessment Mark! Just one last thing, I am missing the pseudocode for # 3 in code challenges. Please just push up those changes and I can finalize your assessment review.

**Review by triciasykes — APPROVED** [review]
great work Mark!

**Pull request merged**